bot.py

```diff
@@ -89,29 +92,99 @@ def __init__(self, guild, loop, dispatcher_user):
+            guarded_by=self.command_lock,
+            help_message="Pause current song",
+        )
+        self.register_command(
+            "resume",
+            handler=self.resume,
+            guarded_by=self.command_lock,
+            help_message="Resume current song",
+        )
+        self.register_command(
+            "skip",
+            handler=self.skip,
+            guarded_by=self.command_lock,
+            help_message="Skip to next song",
+        )
+        self.register_command(
+            "next",
+            handler=self.skip,
+            guarded_by=self.command_lock,
+            help_message="Skip to next song",
+        )
+        self.register_command(
+            "disconnect",
+            handler=self.disconnect,
+            guarded_by=self.command_lock,
+            help_message="Disconnect from the current voice client",
+        )
+        self.register_command(
+            "clear",
+            handler=self.clear_queue,
+            guarded_by=self.command_lock,
+            help_message="Clear the current queue",
+        )
+        self.register_command(
+            "queue",
+            handler=self.show_queue,
+            guarded_by=self.command_lock,
+            help_message="Show the current queue",
+        )
+        self.register_command(
+            "nowplaying",
+            handler=self.show_current,
+            help_message="Show the currently playing song",
         )
-
-        self.register_command("hello", handler=self.hello)
-        self.register_command("countdown", handler=self.countdown)
         self.register_command(
-            "dinkster", handler=self.dinkster, guarded_by=self.command_lock
+            "source",
+            handler=self.show_source,
+            help_message="Show the link to the currently playing song",
+        )
+        self.register_command(
+            "help",
+            handler=self.show_help,
+            help_message="Show this help message or help for given command",
+            argument_name="command",
         )
-        self.register_command("joke", handler=self.joke)
 
-    def register_command(
-        self, command_name, handler=None, guarded_by: asyncio.Lock = None
+        self.register_command("hello", handler=self.hello, help_message="Say hello")
+        self.register_command(
+            "countdown",
+            handler=self.countdown,
+            help_message="Count down from 10 and explode",
+        )
+        self.register_command(
+            "dinkster",
+            handler=self.dinkster,
+            guarded_by=self.command_lock,
+            help_message="Ring the dinkster in your voice channel",
+        )
+        self.register_command("joke", handler=self.joke, help_message="Tell a joke")
+
+    def register_command(  # pylint: disable=too-many-arguments
+        self,
+        command_name,
+        handler=None,
+        guarded_by: asyncio.Lock = None,
+        help_message: str = "",
```

Comment:
Optional suggestion: Make the help_message a required argument.

Reply:
Done